File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int max_height = 0;
    int leftMost = 0;
    
    private void recursion(TreeNode root , int height) {
        if(root != null) {
            height++;
            if(max_height < height) {
                max_height = height;
                leftMost = root.val;
            }
            recursion(root.left , height);
            recursion(root.right , height);
        }
    }
    
    public int findBottomLeftValue(TreeNode root) {
        max_height = 0;
        recursion(root , 0);
        return leftMost;
    }
}

File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int max_height = 0;
    int leftMost = NULL;
    void recursion(TreeNode* root , int height) {
         if(root != NULL) {
             height++;
             if(max_height < height) {
                 max_height = height;
                 leftMost = root->val;
             }
             recursion(root->left , height);
             recursion(root->right , height);
         }
    }
    
    int findBottomLeftValue(TreeNode* root) {
        max_height = 0;
        recursion(root , 0);
        return leftMost;
    }
};